[BUG] - Unhandled Exception: CameraException(404, No barcode view found) after close

**Describe the bug**
I exactly followed your introduction from the README.md to scan some qr codes. 
When I call `cancel` from the `QRViewController` instance I got the following exception:  

```
D/CameraInstance(26864): Closing camera
D/CameraPreview(26864): pause()
E/flutter (26864): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: CameraException(404, No barcode view found)
E/flutter (26864): #0      QRViewController.stopCamera (package:qr_code_scanner/src/qr_code_scanner.dart:268:7)
E/flutter (26864): <asynchronous suspension>
E/flutter (26864): #1      QRViewController.dispose (package:qr_code_scanner/src/qr_code_scanner.dart:294:5)
E/flutter (26864): #2      _QrCodeScannerScreenState.dispose (package:appella_ui/screen/qr_code_scanner/qr_code_scanner_screen.dart:193:24)
E/flutter (26864): #3      StatefulElement.unmount (package:flutter/src/widgets/framework.dart:4854:12)
E/flutter (26864): #4      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1998:13)
E/flutter (26864): #5      _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1996:7)
E/flutter (26864): #6      SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:6105:14)
```

The code from the widget: 

``` dart

class QrCodeResult {
  final String username;
  final String password;

  QrCodeResult({
    @required this.username,
    @required this.password,
  });

  factory QrCodeResult.fromMap(Map<String, dynamic> map) {
    return QrCodeResult(
      username: map['username'] as String,
      password: map['password'] as String,
    );
  }
}

class QrCodeScannerScreen extends StatefulWidget {
  @override
  _QrCodeScannerScreenState createState() => _QrCodeScannerScreenState();
}

class _QrCodeScannerScreenState extends State<QrCodeScannerScreen> {
  final GlobalKey _qrViewKey = GlobalKey<State<StatefulWidget>>();

  QRViewController _qrViewController;

  StreamSubscription<Barcode> _qrScanSubscription;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Padding(
          padding: const EdgeInsets.symmetric(horizontal: 16),
          child: QRView(
            key: _qrViewKey,
            onQRViewCreated: _onQRViewCreated,
          ),
        ),
      ),
    );
  }

  //From the documentation: https://pub.dev/packages/qr_code_scanner
  @override
  void reassemble() {
    super.reassemble();
    if (Platform.isAndroid) {
      _qrViewController.pauseCamera();
    } else if (Platform.isIOS) {
      _qrViewController.resumeCamera();
    }
  }

  void _onQRViewCreated(QRViewController controller) {
    setState(() => _qrViewController = controller);
    _qrScanSubscription = _qrViewController.scannedDataStream
        .where((event) => event.format == BarcodeFormat.qrcode)
        .distinct((l, r) => l.code == r.code)
        .listen(_onScanResult);
  }

  void _onScanResult(Barcode barcode) {
    final scanData = barcode.code;
    try {
      final map = json.decode(scanData) as Map<String, Object>;

      final result = QrCodeResult.fromMap(map);

      _qrViewController?.dispose();
      _qrScanSubscription?.cancel();

      Navigator.of(context).pop(result);
    } catch (e) {
      print(e);
    }
  }

  @override
  void dispose() {
    _qrScanSubscription?.cancel();
    _qrViewController?.dispose();
    super.dispose();
  }
}

```


**Flutter information**

[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.7 19H2 darwin-x64, locale de-DE)
    • Flutter version 1.22.6 at /Users/stefanschaller/Documents/Development/Tapped/appella_app/.flutter
    • Framework revision 9b2d32b605 (vor 3 Wochen), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/stefanschaller/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: /Users/stefanschaller/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Users/stefanschaller/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.49.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

**Device (please complete the following information):**
 - Device: Google Pixel 2
 - OS: Android 11

**Additional context**
Add any other context about the problem here.

I'm using the version 0.3.2.

In 0.3.3 I got the following error message during the build on android: 

```
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.2/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (13, 7): Redeclaration: FlutterQrPlugin
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.2/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (20, 7): Redeclaration: QRView
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.2/android/src/main/kotlin/net/touchcapture/qr/flutterqr/Shared.kt: (5, 8): Redeclaration: Shared
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (13, 7): Redeclaration: FlutterQrPlugin
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (19, 31): Cannot access 'onAttachedToV1': it is private in 'FlutterQrPlugin'
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (24, 16): Unresolved reference: registrar
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (48, 16): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (53, 16): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (58, 16): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/FlutterQrPlugin.kt: (63, 16): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (23, 7): Redeclaration: QRView
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (33, 20): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (34, 20): Unresolved reference: binding
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (37, 20): Unresolved reference: registrar
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/QRView.kt: (38, 20): Unresolved reference: registrar
e: /Users/stefanschaller/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.3.3/android/src/main/kotlin/net/touchcapture/qr/flutterqr/Shared.kt: (7, 8): Redeclaration: Shared

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':qr_code_scanner:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 24s
```

--- a/lib/src/qr_code_scanner.dart
+++ b/lib/src/qr_code_scanner.dart
@@ -1,4 +1,5 @@
 import 'dart:async';
+import 'dart:io';
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
@@ -311,7 +312,7 @@ class QRViewController {
 
   /// Stops the camera and disposes the barcode stream.
   void dispose() {
-    stopCamera();
+    if (Platform.isIOS) stopCamera();
     _scanUpdateController.close();
   }
 

